<a href="https://colab.research.google.com/github/pgallino/Organizacion-de-datos-TP1-1C2023/blob/main/pandas2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


PANDAS

2) ¿Qué porcentaje de las versiones actuales son páginas que se editaron una sola vez? (⭐)

Leo el csv, ya lo vi previamente y solo necesito las columnas correspondientes a id, revision_id y parent_revision.

In [ ]:
df_contents = pd.read_csv('/content/drive/MyDrive/Wikipedia dataset 2c2021/contents.csv')
df_contents.head(10)

title  id  namespace  revision_id  \
0  Wikipedia:Artículos solicitados   5          4    132533307   
1                          Andorra   7          0    138035057   
2                        Argentina  10          0    137966826   
3             Geografía de Andorra  15          0    137986788   
4            Demografía de Andorra  17          0    130869554   
5        Comunicaciones de Andorra  18          0    132654704   
6                   Artes visuales  19          0    138035407   
7                      Agricultura  20          0    138023408   
8             Astronomía galáctica  22          0    134653454   
9                            ASCII  23          0    137833249   

   parent_revision    revision_timestamp revisor_username  revisor_id  \
0      132516631.0  2021-01-19T18:40:34Z        Metrónomo   1708233.0   
1      138027475.0  2021-08-31T18:27:01Z              NaN         NaN   
2      137825165.0  2021-08-28T17:17:54Z        Traitrich   4642283.0   
3      133666383.0  2021-08-29T17:16:00Z            Lojwe   5338508.0   
4      129413188.0  2020-11-12T19:56:15Z           Jialxv   3615366.0   
5      125753476.0  2021-01-24T13:33:34Z    PatriHorrillo    689866.0   
6      138035406.0  2021-08-31T18:45:28Z          SeroBOT   4980693.0   
7      137909816.0  2021-08-31T04:43:18Z         Arctos71   6331749.0   
8      120853916.0  2021-04-10T07:54:41Z          LOLAOMI   6048511.0   
9      137833185.0  2021-08-22T16:18:01Z             Jkbw    936194.0   

                               revisor_ip  \
0                                     NaN   
1  2800:CD0:FC1B:A900:C5A4:9E81:7175:2ED2   
2                                     NaN   
3                                     NaN   
4                                     NaN   
5                                     NaN   
6                                     NaN   
7                                     NaN   
8                                     NaN   
9                                     NaN   

                                     revisor_comment  
0                                     Así esta mejor  
1                                   Añadir contenido  
2  Reemplazos con [[Usuario:Benjavalero/Replacer|...  
3  Reemplazos con [[Usuario:Benjavalero/Replacer|...  
4  Correcciones ortográficas con [[Usuario:Benjav...  
5  Correcciones ortográficas con [[Usuario:Benjav...  
6  Revertidos los cambios de [[Special:Contributi...  
7                                                NaN  
8  Correcciones ortográficas con [[Usuario:Benjav...  
9  Revertidos los cambios de [[Special:Contributi...

Si ya tuvo una revisión y no tiene parent_revision, quiere decir que solo tuvo una revisión. Agrego una columna indicando true, si tuvo una sola revisión.

In [ ]:
total = df_contents['revision_id'].count()
una_rev = total - df_contents['parent_revision'].count()
porcentaje = (una_rev / total) * 100

In [ ]:
porcentaje

35.89446111044963

ACA CORREGÍ, CON DOS COUNT LO RESOLVÍ.

In [ ]:
df_contents['1_revision'] = (~(df_contents['revision_id'].isnull()) & (df_contents['parent_revision'].isnull()))

Cuento todos los que tuvieron una sola revisión y los que no.



In [ ]:
df_una_revision = df_contents.groupby(["1_revision"]).count()

Me queda un dataframe, con índice 1_revisión (true/false) indicando la cantidad de páginas con una revisión y la cantidad que no tienen o tienen más de una. Esta cantidad puedo verla en la columna 'id'

In [ ]:
df_una_revision

id  revision_id  parent_revision
1_revision                                       
False       2648946      2648946          2648946
True        1483218      1483218                0

Para limpiar el dataframe, me quedo con la columna que me interesa, y cambio el nombre de la columna.

In [ ]:
df_una_revision = df_una_revision[['id']]
df_una_revision = df_una_revision.rename(columns= {'id':'cantidad'})
df_una_revision

cantidad
1_revision          
False        2648946
True         1483218

Calculo el porcentaje

In [ ]:
df_una_revision['porcentual'] = df_una_revision['cantidad'] / df_una_revision['cantidad'].sum()
df_una_revision

cantidad  porcentual
1_revision                      
False        2648946    0.641055
True         1483218    0.358945

Finalmente puedo ver, que el porcentaje de páginas con una revisión es del 35,89% aproximadamente, mientras que las que no tienen una sola revisión es del 64% aproximadamente.